### PR TITLE
Adds markdown-link-check and fixes a broken link

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Note the image here is a bad link, but I don't have content for it:

---

### 2.4 Prompt Format
You can even run `llama model prompt-format` see all of the templates and their tokens:

```
llama model prompt-format -m Llama3.2-3B-Instruct
```
<p align="center">
image
</p>

### CODE REVIEW

File: llama_stack/providers/registry/inference.py [hunk 1]
```diff
@@ -98,9 +98,7 @@ def available_providers() -> List[ProviderSpec]:
             api=Api.inference,
             adapter=AdapterSpec(
                 adapter_id="bedrock",
-                pip_packages=[
-                    "boto3"
-                ],
+                pip_packages=["boto3"],
```
Comment (russellb) [inline llama_stack/providers/registry/inference.py:101]: It looks like you have a couple of Python changes (this one and another in augment_messages.py) that are unrelated to markdown linting.

**Review by russellb — CHANGES_REQUESTED** [review]

File: llama_stack/providers/registry/inference.py (hunk 1 above)
Comment (ashwinb) [reply llama_stack/providers/registry/inference.py:101]: Yeah I think it just feels like they ran pre-commit and ended up doing formatting. Seems OK.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -51,3 +51,9 @@ repos:
 #   hooks:
 #     - id: pydoclint
 #       args: [--config=pyproject.toml]
+
+- repo: https://github.com/tcort/markdown-link-check
+  rev: v3.11.2
+  hooks:
+    - id: markdown-link-check
+      args: ['--quiet']
```
[the remark this answers is not in the document]
Comment (ashwinb) [reply .pre-commit-config.yaml:59]: This feels like it _is_ related to the PR? The title says they are adding a link check.

**Review by ashwinb — APPROVED** [review]
lgtm thank you!

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (russellb) [reply .pre-commit-config.yaml:59]: Sorry, my comment wasn't clear. I mean *MY* comment isn't related to the PR.  In other words, I'm not suggesting this PR should change, but just sharing a thought for the future (that we should also run these checks in CI)

File: llama_stack/providers/registry/inference.py (hunk 1 above)
Comment (russellb) [reply llama_stack/providers/registry/inference.py:101]: Ah, makes sense.

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashwinb) [reply .pre-commit-config.yaml:59]: Ohhh! yes, absolutely. Let's file an issue for it. We have CI landing shortly on this repository but having an issue to make sure we don't forget this specific bit would be useful.

[the remark this answers is not in the document]
Comment (russellb) [reply .pre-commit-config.yaml:59]: Filed #172 to come back and see if all the pre-commit checks are running in CI at some point

**Review by russellb — APPROVED** [review]
lgtm

**Comment by codefromthecrypt** [comment]
thanks for resolving this on own! cheers
